<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>NewDatabaseWizard</class>
 <widget class="QWizard" name="NewDatabaseWizard">
  <property name="windowModality">
   <enum>Qt::WindowModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>500</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Create new database</string>
  </property>
  <widget class="QWizardPage" name="titleWizardPage">
   <property name="title">
    <string>Choose a database name</string>
   </property>
   <layout class="QFormLayout" name="formLayout">
    <property name="fieldGrowthPolicy">
     <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
    </property>
    <item row="1" column="0">
     <widget class="QLabel" name="titleLabel">
      <property name="text">
       <string>Title:</string>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QLineEdit" name="titleLineEdit"/>
    </item>
    <item row="0" column="0" colspan="2">
     <widget class="QLabel" name="nameInfoLabel">
      <property name="text">
       <string>Give your database a meaningful name. This does not necessarily have to be the same as the file name. &lt;b&gt;Example:&lt;/b&gt; Joe's Passwords</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWizardPage" name="passwordWizardPage">
   <property name="title">
    <string>Choose a database password</string>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="2" column="0">
     <widget class="QLabel" name="passwordLabel">
      <property name="text">
       <string>Password:</string>
      </property>
     </widget>
    </item>
    <item row="2" column="2">
     <widget class="QLineEdit" name="passwordLineEdit">
      <property name="echoMode">
       <enum>QLineEdit::Password</enum>
      </property>
     </widget>
    </item>
    <item row="2" column="3">
     <widget class="QToolButton" name="revealPasswordToolButton">
      <property name="text">
       <string>...</string>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
      <property name="checked">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="0" column="0" colspan="4">
     <widget class="QLabel" name="passwordInfoLabel">
      <property name="text">
       <string>&lt;p&gt;The database password is used to encrypt all of your other passwords and data you store in your database. Make sure to choose a strong password, and do not reveal it to others or write it down. You may change your password at any time by going to &lt;b&gt;File &amp;gt; Change Database Password...&lt;/b&gt;&lt;/p&gt;

&lt;p&gt;While Silverlock does not enforce password complexity requirements for flexibility reasons, we recommend that your password contain at least one uppercase letter, lowercase letter, number, and symbol, and be at least 10 characters in length.&lt;/p&gt;

&lt;p&gt;The progress bar below can help you to determine how secure your password is. It is not an absolute, but rather a very rough guideline.&lt;/p&gt;</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="4" column="2">
     <widget class="QProgressBar" name="passwordProgressBar">
      <property name="value">
       <number>0</number>
      </property>
      <property name="textVisible">
       <bool>false</bool>
      </property>
     </widget>
    </item>
    <item row="5" column="0" colspan="4">
     <widget class="QLabel" name="passwordWarningLabel">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string>Note that if you lose or forget your database password,&lt;br /&gt;&lt;b&gt;you will not be able to access your database!&lt;/b&gt;</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="1" column="2">
     <widget class="QLabel" name="warningLabel">
      <property name="text">
       <string>&lt;font color=&quot;red&quot;&gt;For security purposes, it is not recommended to use a blank password. Using a blank password is the same as using no password at all.&lt;/color&gt;</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWizardPage" name="wizardPage">
   <property name="title">
    <string>Database creation successful</string>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QLabel" name="continueLabel">
      <property name="text">
       <string>&lt;p&gt;The creation of your database was successful! You may now close this wizard by clicking &lt;b&gt;Finish&lt;/b&gt; and then begin to add groups and password entries to your database.&lt;/p&gt;

&lt;p&gt;If you are not sure how to use a feature or are having trouble, press &lt;b&gt;F1&lt;/b&gt; to access the help menu or visit &lt;a href=&quot;http://www.petroules.com/&quot;&gt;www.petroules.com&lt;/a&gt; for assistance.&lt;/p&gt;</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
      <property name="openExternalLinks">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>NewDatabaseWizard</sender>
   <signal>accepted()</signal>
   <receiver>NewDatabaseWizard</receiver>
   <slot>databaseCreated()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>249</x>
     <y>179</y>
    </hint>
    <hint type="destinationlabel">
     <x>249</x>
     <y>179</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>passwordLineEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>NewDatabaseWizard</receiver>
   <slot>passwordChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>262</x>
     <y>187</y>
    </hint>
    <hint type="destinationlabel">
     <x>249</x>
     <y>179</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>revealPasswordToolButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>NewDatabaseWizard</receiver>
   <slot>passwordVisibilityToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>478</x>
     <y>187</y>
    </hint>
    <hint type="destinationlabel">
     <x>249</x>
     <y>179</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>databaseCreated()</slot>
  <slot>passwordVisibilityToggled(bool)</slot>
  <slot>passwordChanged(QString)</slot>
 </slots>
</ui>
